src/factories/flowRunArtifacts.ts

```diff
@@ -143,46 +142,47 @@ export async function flowRunArtifactsFactory() {
   }
 
   async function handleCollision(
-    prevItem: FlowRunArtifactFactory | ArtifactClusterFactory,
-    currentItem: FlowRunArtifactFactory | ArtifactClusterFactory,
+    prevItem: FlowRunArtifactFactory,
+    currentItem: FlowRunArtifactFactory,
   ): Promise<void> {
-    const isPrevItemCluster = 'getCurrentData' in prevItem
-    const isCurrentCluster = 'getCurrentData' in currentItem
-
-    const prevDate = isPrevItemCluster ? prevItem.getCurrentData()?.date : prevItem.data.created
-    const prevIds = isPrevItemCluster ? prevItem.getCurrentData()?.ids : [prevItem.data.id]
-    const currentDate = isCurrentCluster ? currentItem.getCurrentData()?.date : currentItem.data.created
-    const currentIds = isCurrentCluster ? currentItem.getCurrentData()?.ids : [currentItem.data.id]
+    const prevDate = prevItem.getDate()
+    const prevIds = prevItem.getIds()
+    const currentDate = currentItem.getDate()
+    const currentIds = currentItem.getIds()
 
     if (!prevDate || !currentDate || !prevIds || !currentIds) {
       console.error('flowRunArtifacts: visible item is missing date or ID data')
       return
     }
 
-    let clusterNode: ArtifactClusterFactory
+    let clusterNode: FlowRunArtifactFactory
 
-    if (isPrevItemCluster) {
+    if (prevItem.isCluster) {
       clusterNode = prevItem
-    } else if (isCurrentCluster) {
+    } else if (currentItem.isCluster) {
       clusterNode = currentItem
     } else {
       clusterNode = await getClusterNode()
     }
 
     const ids = [...prevIds, ...currentIds]
-    const date = new Date((prevDate.getTime() + currentDate.getTime()) / 2)
 
-    clusterNode.render({ ids, date })
+    const dates = ids.map(id => artifacts.get(id)?.getDate())
+    const minDate = new Date(Math.min(...dates.map(date => date?.getTime() ?? 0)))
+    const maxDate = new Date(Math.max(...dates.map(date => date?.getTime() ?? 0)))
```

Comment:
There are `min` and `max` utilities in `date-fns` that you could use here.

Reply:
In [a PR](https://github.com/PrefectHQ/graphs/pull/410) into this one I actually change this up a bit, might as well review it there if you don't mind.